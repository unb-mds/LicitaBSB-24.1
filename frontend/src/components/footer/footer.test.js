import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Footer from './';
import { BrowserRouter } from 'react-router-dom';

describe('Deve renderizar os links para as rotas: ', () => {
  test('Licitações', () => {
    render(<BrowserRouter>
      <Footer />
    </BrowserRouter>);
    const linkRoute = screen.getByText('Licitações');
    expect(linkRoute).toBeInTheDocument();
  });
  test('SobreLicitacao', () => {
    render(<BrowserRouter>
      <Footer />
    </BrowserRouter>);
    const linkRoute = screen.getByText('Sobre as Licitações');
    expect(linkRoute).toBeInTheDocument();
  });
  test('sobrenos', () => {
    render(<BrowserRouter>
      <Footer />
    </BrowserRouter>);
    const linkRoute = screen.getByText('Sobre a Equipe');
    expect(linkRoute).toBeInTheDocument();
  });
  test('Dashboard', () => {
    render(<BrowserRouter>
      <Footer />
    </BrowserRouter>);
    const linkRoute = screen.getByText('Dashboard');
    expect(linkRoute).toBeInTheDocument();
  });
});

test('Deve renderizar a lista de links presentes no componente', () => {
  render(<BrowserRouter>
    <Footer />
  </BrowserRouter>);
  const listaDeLinks = screen.getAllByRole('listitem');
  expect(listaDeLinks).toHaveLength(4);
});

test('Deve renderizar o snapshot da lista de links presentes no componente', () => {
  render(<BrowserRouter>
    <Footer />
  </BrowserRouter>);
  const listaDeLinks = screen.getAllByRole('listitem');
  expect(listaDeLinks).toMatchSnapshot();
});
