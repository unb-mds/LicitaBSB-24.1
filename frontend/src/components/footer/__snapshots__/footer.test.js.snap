// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o snapshot da lista de links presentes no componente 1`] = `
Array [
  <li>
    <a
      href="/licitacoes"
    >
      <p>
        Licitações
      </p>
    </a>
  </li>,
  <li>
    <a
      href="/sobrelicitacao"
    >
      <p>
        Sobre as Licitações
      </p>
    </a>
  </li>,
  <li>
    <a
      href="/sobrenos"
    >
      <p>
        Sobre a Equipe
      </p>
    </a>
  </li>,
  <li>
    <a
      href="/dashboard"
    >
      <p>
        Dashboard
      </p>
    </a>
  </li>,
]
`;
